### PR TITLE
W-035967 - Update descriptions in cumulusci.yml and add additional qa metadata & configuration

### PR DESCRIPTION
**Pull request by xinnanlisf**
This internal task adds page layouts, search layouts, and methods to enable custom settings for the EDA qa_org flow. This enables our QE team to use specific customizations when spinning up a testing org. This task also organizes the related descriptions in the cumulusci.yml file. 

# Critical Changes

# Changes

# Issues Closed

# New Metadata

# Deleted Metadata

# Testing Notes

### CODE REVIEW

File: cumulusci.yml [hunk 1]
```diff
@@ -87,6 +87,7 @@ tasks:
             output: robot/EDA/doc/Keywords.html
 
     update_admin_profile:
+        description: Sets record type visibilities for System Administrative profile
```
Comment (jstvz) [inline cumulusci.yml]: I'd like to not overwrite the built-in description. Can we add to it instead? Suggested:

```suggestion
        description: Retrieves, edits, and redeploys the System Administrator profile with full FLS perms for all objects/fields. Also sets record type visibilities.
        group: Custom Tasks
```

Also adding `group:` here to make clear that we've overridden the default task.

File: cumulusci.yml [hunk 2]
```diff
@@ -126,37 +131,25 @@ flows:
...
-                    namespaced_org: True
-                    skip_record_types: True
-
     config_qa:
+        descripton: Configure an org for use as a QA org after package metadata is deployed
         steps:
             3:
                 task: execute_install_apex
             4:
-                task: deploy_dev_config
```
Comment (xinnanlisf) [inline cumulusci.yml]: @jstvz do you think it is ok for us to remove this line? config_qa should only deploy qa config?

File: cumulusci.yml [hunk 3]
```diff
@@ -37,8 +37,8 @@ tasks:
         options:
             path: unpackaged/config/dev_delete
 
-    execute_install_apex:
-        description: Runs most of the install script methods from STG_InstallScript class
+    execute_dev_apex:
```
Comment (xinnanlisf) [inline cumulusci.yml]: @jstvz I renamed it to dev_apex since it feels like it is running apex for dev orgs. We can change it back if you prefer install_apex thou.

File: cumulusci.yml [hunk 4]
```diff
@@ -167,21 +162,6 @@ flows:
                 options:
                     unmanaged: False
 
-    deploy_unmanaged:
```
Comment (xinnanlisf) [inline cumulusci.yml]: @jstvz i have removed these flows too since we are not adding tasks to them.

File: scripts/setup.cls [hunk 5]
```diff
@@ -1,17 +1,46 @@
 public static void qaSetup() {
-  enableCourseConnections();
+    configQACustomSettings();
```
Comment (xinnanlisf) [inline scripts/setup.cls]: @robertbwatson not sure if you are doing the same work to enable settings for dev/qa configs. but fyi..

Comment (robertbwatson) [reply scripts/setup.cls]: @xinnanlisf  No worries - I modified the execute_install_apex task (or execute_dev_apex) in #945 so should be fine here.

File: cumulusci.yml (hunk 3 above)
Comment (robertbwatson) [reply cumulusci.yml]: It is doing most of the install script though, of a new install, so the old name does sort of make sense. It's doing lines 44-63 of STG_InstallScript.cls, minus the scheduling of the scheduled jobs and running the batch email script.

(I don't have a strong opinion either way)

File: cumulusci.yml [hunk 6]
```diff
@@ -167,21 +162,6 @@ flows:
...
-        steps:
-            4:
-                # Disable uninstall_packaged_incremental
-                task: None
-
-    dev_org_namespaced:
```
Comment (robertbwatson) [inline cumulusci.yml]: Wait - I use this one all of the time. This is the flow we call to set up a scratch org with a namespace. I don't think all of the steps are defined here because it's a core CCI flow.

File: cumulusci.yml (hunk 3 above)
Comment (xinnanlisf) [reply cumulusci.yml]: Fair enough, qa config is also running this thus I don't think it makes sense to call it dev_apex either...Ok I will change it back.

File: cumulusci.yml (hunk 6 above)
Comment (xinnanlisf) [reply cumulusci.yml]: Right, many of the flows in this yml file is basically just adding more tasks or simply overriding. For example, dev_org does not even exist in this yml file but it is defined in sfdo tooling yml file.

Comment (robertbwatson) [reply cumulusci.yml]: But why is this being removed? We're modifying step 3 to make sure to pass in namespaced_org: true to the update_admin_profile task - does that not matter? (I don't quite understand this but I also don't want the flow to suddenly break)

Comment (xinnanlisf) [reply cumulusci.yml]: I see so it is overriding the options for #3 task. I will add them back then!

Comment (robertbwatson) [reply cumulusci.yml]: I think so but I don't know for certain... would be something that @jstvz could help explain :)

**Comment by xinnanlisf** [comment]
@robertbwatson this PR should be ready for review. Since you are ready looking into this, would you mind taking a review?

Also, @jstvz I added a few things after you approved this PR. Would you mind giving a sanity check too?

Thanks!

File: cumulusci.yml (hunk 2 above)
Comment (jstvz) [reply cumulusci.yml]: There's some overlap here, but I lean towards deploying it as part of the QA config. This is for two reasons: first, because new schema will often be updated here first, and second, to avoid having to duplicate sets of metadata.

That said, why did you want to remove it?

File: cumulusci.yml [hunk 7]
```diff
@@ -126,37 +133,25 @@ flows:
...
                     unmanaged: False
 
-    config_packaging:
-        steps:
-            1:
-                # Configure update_admin_profile
-                options:
-                    managed: True
-                    namespaced_org: True
-                    skip_record_types: True
```
Comment (jstvz) [inline cumulusci.yml]: Can we restore this block, please? Recall that we're overriding `update_admin_profile` with a custom task that includes the `skip_record_types` option.

File: cumulusci.yml [hunk 8]
```diff
@@ -126,37 +133,25 @@ flows:
                 task: deploy_dev_config
 
     config_managed:
+        description: Configure an org for use as a dev org after package metadata is deployed.
         steps:
-            2:
-                # Configure update_admin_profile
-                options:
-                    managed: True
```
Comment (jstvz) [inline cumulusci.yml]: This one is okay to remove, since it's duplicating the built-in option.

File: cumulusci.yml [hunk 9]
```diff
@@ -48,69 +48,76 @@ tasks:
...
                 task: uninstall_packaged
 
     config_apextest:
+        description: Configure an org to run apex tests after package metadata is deployed.
         steps:
             3:
                 task: execute_install_apex
 
     config_dev:
+        description: Configure an org for use as a dev org after package metadata is deployed.
```
Comment (jstvz) [inline cumulusci.yml]: I'm guessing that you're adding descriptions here so that the configuration file is easier to understand. I think that makes sense, but should we consider that by doing so we run the risk of them going out of sync? Perhaps for those built-in tasks/flows that you're adding descriptions for we could add comments instead?

File: robot/EDA/resources/EDA.py [hunk 10]
```diff
@@ -1,12 +1,16 @@
 import logging
 import time
+import datetime
```
Comment (jstvz) [inline robot/EDA/resources/EDA.py]: These imports are unused. Others below on L13 are duplicated. Unless you intended to update these files, they should be reverted to the version on `master`.

File: robot/EDA/tests/browser/pe/program_enrollments.robot [hunk 11]
```diff
@@ -2,6 +2,7 @@
 
 Resource        robot/EDA/resources/EDA.robot
 Library         robot/EDA/resources/EDA.py
+Library         SeleniumLibrary
```
Comment (jstvz) [inline robot/EDA/tests/browser/pe/program_enrollments.robot]: This change is unrelated and should be included in a separate PR.

File: scripts/setup.cls [hunk 12]
```diff
@@ -1,17 +1,46 @@
...
+    settings.put('%%%NAMESPACE%%%Enable_Course_Connections__c', true);
+    settings.put('%%%NAMESPACE%%%Student_RecType__c', rtInfosByName.get('Student').getRecordTypeId());
+    settings.put('%%%NAMESPACE%%%Faculty_RecType__c', rtInfosByName.get('Faculty').getRecordTypeId());
+}
 
-  %%%NAMESPACE%%%Hierarchy_Settings__c settings = %%%NAMESPACE%%%Hierarchy_Settings__c.getOrgDefaults();
+//Enable Specify Role For Created Affiliations
+private static void enableSpecifyRoleForCreatedAffiliations(%%%NAMESPACE%%%Hierarchy_Settings__c settings) {   
+    if (settings.%%%NAMESPACE%%%Affl_ProgEnroll_Set_Role__c == null 
+        || settings.%%%NAMESPACE%%%Enable_New_Preferred_Phone_Sync__c != true) {
```
Comment (jstvz) [inline scripts/setup.cls]: This is a different field from the null check in this conditional. Is that intentional?

File: cumulusci.yml (hunk 2 above)
Comment (xinnanlisf) [reply cumulusci.yml]: @jstvz that makes a lot of sense. I was under the impression that dev config is only used for dev org and qa org should use its own config. I will go ahead and add it back.

File: robot/EDA/resources/EDA.py (hunk 10 above)
Comment (xinnanlisf) [reply robot/EDA/resources/EDA.py]: Yeah i think these changes are due to relEng fix on robot tests. I will remove them. Thanks!

File: cumulusci.yml (hunk 9 above)
Comment (xinnanlisf) [reply cumulusci.yml]: That is a great idea!

File: cumulusci.yml [hunk 13]
```diff
@@ -38,7 +38,7 @@ tasks:
             path: unpackaged/config/dev_delete
 
     execute_install_apex:
-        description: Runs most of the install script methods from STG_InstallScript class
+        # description: Runs anonymous apex for dev to config settings, insert Affiliation mappings, insert Relationship lookups and insert default tdtm trigger handlers.
```
Comment (jstvz) [inline cumulusci.yml]:
```suggestion
        description: Runs anonymous apex for dev to config settings, insert Affiliation mappings, insert Relationship lookups and insert default tdtm trigger handlers.
```

File: cumulusci.yml [hunk 14]
```diff
@@ -48,69 +48,76 @@ tasks:
                 TDTM_Global_API.setTdtmConfig( TDTM_Global_API.getTdtmConfig() );
 
     execute_qa_apex:
-        description: Runs Apex setup for QA orgs.
+        # description: Runs anonymous apex for qa to enable course connections.
```
Comment (jstvz) [inline cumulusci.yml]:
```suggestion
        description: Runs anonymous apex for qa to enable course connections.
```

File: cumulusci.yml [hunk 15]
```diff
@@ -146,17 +150,17 @@ flows:
                     skip_record_types: True
 
     config_qa:
+        # descripton: Configure an org for use as a QA org after package metadata is deployed.
         steps:
             3:
                 task: execute_install_apex
             4:
-                task: deploy_dev_config
```
Comment (jstvz) [inline cumulusci.yml]: I don't think this made it back in yet

File: cumulusci.yml (hunk 14 above)
Comment (robertbwatson) [reply cumulusci.yml]: I think the proposed new description here is invalid. It does other things now besides enabling course connections, right?

File: unpackaged/config/qa/objects/___NAMESPACE___Course_Offering__c.object [hunk 16]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<CustomObject xmlns="http://soap.sforce.com/2006/04/metadata">
+    <searchLayouts>
+        <searchResultsAdditionalFields>Course__c</searchResultsAdditionalFields>
```
Comment (robertbwatson) [inline unpackaged/config/qa/objects/___NAMESPACE___Course_Offering__c.object]: Do these need the %%%namespace%%%?

File: unpackaged/config/qa/objects/___NAMESPACE___Plan_Requirement__c.object [hunk 17]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<CustomObject xmlns="http://soap.sforce.com/2006/04/metadata">
+    <searchLayouts>
+        <searchResultsAdditionalFields>Plan_Requirement__c</searchResultsAdditionalFields>
```
Comment (robertbwatson) [inline unpackaged/config/qa/objects/___NAMESPACE___Plan_Requirement__c.object]: Do these need the %%%namespace%%%?

File: unpackaged/config/qa/objects/___NAMESPACE___Term_Grade__c.object [hunk 18]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<CustomObject xmlns="http://soap.sforce.com/2006/04/metadata">
+    <searchLayouts>
+        <searchResultsAdditionalFields>Course_Connection__c</searchResultsAdditionalFields>
```
Comment (robertbwatson) [inline unpackaged/config/qa/objects/___NAMESPACE___Term_Grade__c.object]: Do these need the %%%namespace%%%?

**Comment by robertbwatson** [comment]
@xinnanlisf  Also, can you add a description to this PR? It can go above Critical Changes so that it doesn't get into the release notes, but will be good for our reference later.

**Comment by xinnanlisf** [comment]
@jstvz @robertbwatson should be good now!

File: cumulusci.yml [hunk 19]
```diff
@@ -48,69 +48,76 @@ tasks:
...
 
     robot_libdoc:
+        # description: Generates documentation for project keyword files.
         options:
             path: robot/EDA/resources/EDA.py,robot/EDA/resources/EDA.robot
             output: robot/EDA/doc/Keywords.html
 
     update_admin_profile:
+        description: Retrieves, edits, and redeploys the System Administrator profile with full FLS perms for all objects/fields. Also sets record type visibilities.
+        group: Custom Tasks
```
Comment (robertbwatson) [inline cumulusci.yml]: This is already listed under the "Salesforce Metadata" group - are we sure we want to move it to a new group?

File: cumulusci.yml [hunk 20]
```diff
@@ -48,69 +48,76 @@ tasks:
...
         class_path: tasks.salesforce.UpdateAdminProfile
```
Comment (robertbwatson) [inline cumulusci.yml]: Also, this isn't a custom task, I don't think - should we actually be overriding the description here? I think we're just overriding the default behavior in tasks/salesforce.py

File: unpackaged/config/qa/objects/Contact.object [hunk 21]
```diff
@@ -19,4 +19,27 @@
...
+        <lookupPhoneDialogsAdditionalFields>CONTACT.PHONE3</lookupPhoneDialogsAdditionalFields>
+        <lookupPhoneDialogsAdditionalFields>CONTACT.PHONE4</lookupPhoneDialogsAdditionalFields>
+        <lookupPhoneDialogsAdditionalFields>CONTACT.PHONE5</lookupPhoneDialogsAdditionalFields>
+        <lookupPhoneDialogsAdditionalFields>CONTACT.PHONE6</lookupPhoneDialogsAdditionalFields>
+        <lookupPhoneDialogsAdditionalFields>WorkPhone__c</lookupPhoneDialogsAdditionalFields>
+        <searchResultsAdditionalFields>FULL_NAME</searchResultsAdditionalFields>
+        <searchResultsAdditionalFields>ACCOUNT.NAME</searchResultsAdditionalFields>
+        <searchResultsAdditionalFields>CONTACT.EMAIL</searchResultsAdditionalFields>
+        <searchResultsAdditionalFields>Primary_Academic_Program__c</searchResultsAdditionalFields>
+        <searchResultsAdditionalFields>Primary_Language__c</searchResultsAdditionalFields>
```
Comment (robertbwatson) [inline unpackaged/config/qa/objects/Contact.object]: Here's another one missing %%%NAMESPACE%%%, I think -- this is packaged (Primary_Academic_Program__c is unpackaged, though)

File: unpackaged/config/qa/objects/___NAMESPACE___Course_Offering__c.object [hunk 22]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<CustomObject xmlns="http://soap.sforce.com/2006/04/metadata">
+    <searchLayouts>
+        <searchResultsAdditionalFields>%%%namespace%%%Course__c</searchResultsAdditionalFields>
```
Comment (robertbwatson) [inline unpackaged/config/qa/objects/___NAMESPACE___Course_Offering__c.object]: @xinnanlisf @jstvz For this & the other files - does casing matter? (Does it need to be %%%NAMESPACE%%%?)

Comment (davisagli) [reply unpackaged/config/qa/objects/___NAMESPACE___Course_Offering__c.object]: It does need to be in all caps -- the code that replaces the tokens is here: https://github.com/SFDO-Tooling/CumulusCI/blob/master/cumulusci/utils.py#L184

File: cumulusci.yml (hunk 19 above)
Comment (xinnanlisf) [reply cumulusci.yml]: I think this was @jstvz suggested?

File: cumulusci.yml (hunk 20 above)
Comment (xinnanlisf) [reply cumulusci.yml]: @jstvz too.

File: unpackaged/config/qa/objects/___NAMESPACE___Course_Offering__c.object (hunk 22 above)
Comment (xinnanlisf) [reply unpackaged/config/qa/objects/___NAMESPACE___Course_Offering__c.object]: Thanks @davisagli! We will just capitalize them so that they look consistent!

File: cumulusci.yml (hunk 19 above)
Comment (jstvz) [reply cumulusci.yml]: Yeah, I suggested this to make it clearer that we're overriding the built-in task with a custom one.

File: cumulusci.yml (hunk 20 above)
Comment (jstvz) [reply cumulusci.yml]: Yeah, my view is that if any task depends on a python class in `tasks/` then it's a custom task and should be highlighted as such. I think it's helpful to keep that in mind when doing any troubleshooting or debugging.

File: cumulusci.yml (hunk 19 above)
Comment (robertbwatson) [reply cumulusci.yml]: Thanks for the explanation!

**Comment by xinnanlisf** [comment]
@jstvz i think this PR is ready. Let me know if any other suggested changes!

File: cumulusci.yml [hunk 23]
```diff
@@ -49,69 +49,76 @@ tasks:
...
 
     robot_libdoc:
+        # description: Generates documentation for project keyword files.
         options:
             path: robot/EDA/resources/EDA.py,robot/EDA/resources/EDA.robot
             output: robot/EDA/doc/Keywords.html
 
     update_admin_profile:
+        description: Retrieves, edits, and redeploys the System Administrator profile with full FLS perms for all objects/fields. Also sets record type visibilities.
+        group: Custom Tasks
```
Comment (jstvz) [inline cumulusci.yml]:
```suggestion
        group: "EDA: Custom Tasks"
```

**Comment by dgamboni** [comment]
@juliesnow This is an internal task only; are these release notes OK?

**Comment by spinkelman** [comment]
Merging per @dgamboni.
